front/lib/api/agent_data_sources.ts

```diff
@@ -128,3 +130,53 @@ export async function getDataSourceUsage({
     ],
   });
 }
+
+export async function getDataSourceViewUsage({
+  auth,
+  dataSourceView,
+}: {
+  auth: Authenticator;
+  dataSourceView: DataSourceViewType;
+}): Promise<number> {
+  const owner = auth.workspace();
+
+  // This condition is critical it checks that we can identify the workspace and that the current
+  // auth is a user for this workspace. Checking `auth.isUser()` is critical as it would otherwise
+  // be possible to access data sources without being authenticated.
+  if (!owner || !auth.isUser()) {
+    return 0;
+  }
+
+  return AgentDataSourceConfiguration.count({
+    where: {
+      dataSourceViewId: dataSourceView.id,
+    },
+    include: [
+      {
+        model: DataSourceViewModel,
+        as: "dataSourceView",
+        where: {
+          workspaceId: owner.id,
+        },
+        attributes: [],
+      },
+      {
+        model: AgentRetrievalConfiguration,
+        as: "agent_retrieval_configuration",
```

Comment:
can we stick to camel case?

Reply:
this is how it's defined in https://github.com/dust-tt/dust/blob/3f69681058786326460d008173eff0f9002900ed/front/lib/models/assistant/actions/retrieval.ts#L95  (and so in all other methods in agent_data_sources)

Reply:
Arf 🙄